src/open_clip/hf_model.py

```diff
@@ -235,7 +240,9 @@ def lock(self, unlocked_layers: int = 0, freeze_layer_norm: bool = True):
 
     @torch.jit.ignore
     def set_grad_checkpointing(self, enable=True):
-        self.transformer.gradient_checkpointing_enable()
+        self.transformer._set_gradient_checkpointing(
+            enable=True, gradient_checkpointing_func=checkpoint
+        )
```

Comment:
With this change, _set_gradient_checkpointing calls the function from JinaBertPreTrainedModel [here](https://huggingface.co/jinaai/jina-bert-implementation/blob/main/modeling_bert.py), that needs different arguments.
`def _set_gradient_checkpointing(self, module, value=False):
        if isinstance(module, BertEncoder):
            module.gradient_checkpointing = value`
 I think it is better to keep the gradient_checkpointing_enable() that calls the function from the PreTrainedModel class of transformers library

Reply:
I wanted to use the deepspeed checkpoint function but that is not possible, will revert